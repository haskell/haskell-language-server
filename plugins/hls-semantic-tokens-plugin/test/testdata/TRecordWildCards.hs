{-# LANGUAGE RecordWildCards #-}
module TRecordWildCards where

data Foo = Foo { foo :: Int }
foo1 = Foo 1
unpackFoo :: Foo -> Int
unpackFoo Foo{..} = foo
