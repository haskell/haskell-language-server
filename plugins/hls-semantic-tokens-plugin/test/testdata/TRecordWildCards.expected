4:6-9 TTypeConstructor "Foo"
4:12-15 TDataConstructor "Foo"
4:18-21 TRecordField "foo"
4:25-28 TTypeConstructor "Int"
5:1-5 TVariable "foo1"
5:8-11 TDataConstructor "Foo"
6:1-10 TFunction "unpackFoo"
6:14-17 TTypeConstructor "Foo"
6:21-24 TTypeConstructor "Int"
7:1-10 TFunction "unpackFoo"
7:11-14 TDataConstructor "Foo"
7:21-24 TVariable "foo"
